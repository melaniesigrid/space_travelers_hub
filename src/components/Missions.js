import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions } from '../redux/missions';
import MissionList from './MissionList';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);

  useEffect(() => {
    if (missions.length < 1) {
      dispatch(getMissions());
    }
  }, [missions]);

  return (
    <>
      <br />
      <br />
      <div className="container">
        <div className="table-responsive">
          <table className="table table-striped table-bordered">
            <thead>
              <tr>
                <th scope="col">Mission</th>
                <th scope="col">Description</th>
                <th scope="col">Status</th>
                <th scope="col">.</th>
              </tr>
            </thead>
            <tbody>
              {
                missions.map((mission) => (
                  <MissionList
                    id={mission.mission_id}
                    name={mission.mission_name}
                    description={mission.description}
                    reserved={mission.reserved}
                    key={mission.mission_id}
                  />
                ))

              }
            </tbody>
          </table>
        </div>

      </div>

    </>
  );
};

export default Missions;
