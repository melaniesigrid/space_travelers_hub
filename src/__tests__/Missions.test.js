import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/Missions';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
