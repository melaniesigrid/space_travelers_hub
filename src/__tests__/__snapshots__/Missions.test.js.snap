// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <br />,
  <br />,
  <div
    className="container"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table table-striped table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Mission
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Status
            </th>
            <th
              scope="col"
            >
              .
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
]
`;
